### PR TITLE
[WIP]Explicit dependency for starscream

### PR DESCRIPTION
**Pull request by lightsprint09**
## Whats new
This adds Starscream as an explicit dependencies. Problems with Obj-C are solved by introducing a wrapper around `Starscream.SSLSecurity`

## Motivation
When Starscream releases critical, non breaking changes, there is no need to update Socket.IO

## What is needed to get this into mergeable state.

[ ] Setup Pods dependency for Starscream
[ ] Setup SPM dependency for Starscream

### CODE REVIEW

**Comment by nuclearace** [comment]
Can you try rebasing on development?

File: Source/SocketIO/Util/SSLSecurity.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//
```
Comment (nuclearace) [inline Source/SocketIO/Util/SSLSecurity.swift]: This'll need docstrings eventually.

**Comment by nuclearace** [comment]
@lightsprint09 Any progress on this? I think this will really help ease things.

**Comment by lightsprint09** [comment]
I did not find time. I will try to do it next week.

**Comment by nuclearace** [comment]
@lightsprint09 So I've been playing around with this in [a test branch](https://github.com/socketio/socket.io-client-swift/tree/lightsprint09-feature/explicit-dependency-starscream). I keep getting [this](https://i.imgur.com/0AZVjVk.png). I remember seeing this before when I tried using Carthage but I can never remember how to fix it.

**Comment by nuclearace** [comment]
I figured it out. Needed a copy files phase.

**Comment by nuclearace** [comment]
@lightsprint09 I've fixed the tests and merged these changes in. Feel free to open another PR with anymore changes this might need. I'll have to fix the Podfile to point to the normal upstream Starscream again, but that's not a big deal.

**Pull request closed without merging**